doc/release-notes-27600.md

```diff
@@ -0,0 +1,10 @@
+P2P
+---
+
+- Inbound connections from hosts protected by `whitebind` or `whitelist`
+  permissions will now be more likely to connect even if `maxconnections` is
+  reached and the node's inbound slots are full. Because this is achieved by
+  more aggressively finding a current connection to evict, users should take
+  care only to set these permissions on local ports that attackers can not
```

Comment:
I think "on local ports" is unnecessarily restrictive/scary. I think it's still completely fine to whitelist remote addresses, you mostly just want to avoid ranges, right?

Reply:
@mzumsande what's your opinion on this? Is this PR a big enough change to warrant this kind of warning?

Reply:
Not really sure, I've never looked into these options much and don't know about best practices - I think that if you use `-whitebind` with non-local addresses, you'd at least have to make sure that that address is not self-advertised. I guess that this applies more to `-whitebind` than `-whitelist`, so that the preferred approach in case of a non-local connection would be to use `-whitelist`?
fyi @vasild, do you have an opinion on this?

Reply:
That's a good point, `whitelist` is harder to attack than `whitebind` because the attacker would have to spoof their origin IP repeatedly to fill up your inbounds. If a user `whitebind`'s a port and an attacker figures out that port numnber, they can trivially evict all your other inbounds

Reply:
I agree with what you say above - `whitebind` on publicly accessible address:port with this new permission sounds bad. Similar with `whitelist` - if a range is used.

This PR currently expands the semantic of the `noban` permission. This will affect existent setups that already use it. Would it make sense to introduce a new permisson, separate from `noban`? I mean - now if somebody is running `-whitebind=noban@publicaddr:port` then a bad actor could cause harm on `master`, but even more harm with this PR.

Reply:
@LarryRuane you are correct but what we are changing (in the current state of this branch) is that `noban` nodes can now force disconnection of other peers. Since many users may already have `noban` set, maybe even for a large range of IPs, this branch would introduce a new vulnerability they may not be aware of. Because of that I think it does make more sense to specify a new permission so users can narrow the attack surface